src/VocabularyManager.Core/Interfaces/IStateManager.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace VocabularyManager.Core.Interfaces
+{
+    public interface IStateManager
+    {
+        public void SubscribeOnChangeAction(Action action);
+        public void UnsubscribeOnChangeAction(Action action);
+        public void NotifyStateHasChanged();
```

Comment:
this looks like `Blazor` specific. Why it would be on server side at all?

Reply:
right)